core/templates/pages/contributor-dashboard-page/modal-templates/translation-suggestion-review-modal.component.ts

```diff
@@ -264,43 +317,106 @@ export class TranslationSuggestionReviewModalComponent implements OnInit {
     const response = await this.threadDataBackendApiService.fetchMessagesAsync(
       threadId);
     const threadMessageBackendDicts = response.messages;
-    this.reviewMessage = threadMessageBackendDicts.map(
-      m => ThreadMessage.createFromBackendDict(m))[1].text;
+    let threadMessages = threadMessageBackendDicts.map(
+      m => ThreadMessage.createFromBackendDict(m));
+    // This is to prevent a console error when a contribution
+    // doesn't have a review message. When a contribution has
+    // a review message the second element of the threadMessages
+    // array contains the actual review message.
+    if (threadMessages[1] !== undefined) {
+      this.reviewMessage = threadMessages[1].text;
+    }
   }
 
-  showNextItemToReview(suggestionId: string): void {
-    this.resolvedSuggestionIds.push(this.activeSuggestionId);
-    if (this.lastSuggestionToReview) {
+  refreshModalData(): void {
+    this.activeSuggestion = this.activeContribution.suggestion;
+    if (this.activeContribution.details === null) {
+      return;
+    }
+    this.activeContributionDetails = this.activeContribution.details;
+    this.contextService.setCustomEntityContext(
+      AppConstants.IMAGE_CONTEXT.EXPLORATION_SUGGESTIONS,
+      this.activeSuggestion.target_id);
+    this.subheading = (
+      `${this.activeContributionDetails.topic_name} / ` +
+      `${this.activeContributionDetails.story_title} / ` +
+      `${this.activeContributionDetails.chapter_title}`
+    );
+    // This is needed to re-calculate height of the panels.
+    this.calculatePanelHeight();
+  }
+
+  gotoNextItem(): void {
+    let lastContributionId = this.remainingContributionIds.pop();
+    // If the current item is the last item, do not navigate.
+    if (lastContributionId === undefined) {
+      return;
+    }
+    // This prevents resolved contributions from getting added to the list.
+    if (!this.resolvedSuggestionIds.includes(this.activeSuggestionId)) {
+      this.skippedContributionIds.push(this.activeSuggestionId);
+    }
+
+    this.activeSuggestionId = lastContributionId;
+    this.activeContribution = this.allContributions[
+      lastContributionId];
+
+    this.isLastItem = this.remainingContributionIds.length === 0;
```

Comment:
Do we still need this?

Reply:
Yes, isLastItem and isFirstItem variables are required to be set as they are used in the frontend to disable naviagtion buttons accordingly.